instrumentation/micrometer/micrometer-1.5/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/micrometer/v1_5/OpenTelemetryGauge.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.micrometer.v1_5;
+
+import static io.opentelemetry.javaagent.instrumentation.micrometer.v1_5.Bridging.baseUnit;
+import static io.opentelemetry.javaagent.instrumentation.micrometer.v1_5.Bridging.description;
+import static io.opentelemetry.javaagent.instrumentation.micrometer.v1_5.Bridging.toAttributes;
+
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.Measurement;
+import io.opentelemetry.api.common.Attributes;
+import java.util.Collections;
+import java.util.function.ToDoubleFunction;
+
+final class OpenTelemetryGauge<T> implements Gauge, RemovableMeter {
+
+  private final Id id;
+  private final Attributes attributes;
+  private final AsyncInstrumentRegistry asyncInstrumentRegistry;
+
+  OpenTelemetryGauge(
+      Id id,
+      T obj,
+      ToDoubleFunction<T> objMetric,
+      AsyncInstrumentRegistry asyncInstrumentRegistry) {
+    this.id = id;
+    this.attributes = toAttributes(id.getTags());
+    this.asyncInstrumentRegistry = asyncInstrumentRegistry;
+
+    asyncInstrumentRegistry.buildGauge(
+        id.getName(), description(id), baseUnit(id), attributes, obj, objMetric);
```

Comment:
Could pass `Id` here instead of `name, `description`, and `baseUnit` to reduce the arguments.